plugins/i18n_subsites/i18n_subsites.py

```diff
@@ -270,7 +270,13 @@ def install_templates_translations(generator):
     Only if the 'jinja2.ext.i18n' jinja2 extension is enabled
     the translations for the current DEFAULT_LANG are installed.
     '''
-    if 'jinja2.ext.i18n' in generator.settings['JINJA_EXTENSIONS']:
+    if 'JINJA_ENVIRONMENT' in generator.settings: # pelican 3.7+
+        jinja_extensions = generator.settings['JINJA_ENVIRONMENT'].get(
+            'extensions', [])
+    else:
+        jinja_extensions = generator.settings['JINJA_EXTENSIONS']
+
+    if 'jinja2.ext.i18n' in jinja_extensions:
```

Comment:
Why do we need reverse compat hacks like this if we pinned the pelican version in `requirements.txt`?

I'd always require contributors to use the version of Pelican as defined in `requirements.txt`, ideally in a virtualenv.

First-time init:

    $ python3 -m venv VIRTUALENV
    $ VIRTUALENV/bin/pip install -r requirements.txt

And then before building:

    $ source VIRTUALENV/bin/activate

I think it would be better to upgrade the pelican version in `requirements.txt` and to remove the reverse compat code.

Reply:
definitely looks strange. Sorry, didn't had in mind this one was still around. But it looks like I managed to replicate this one exactly:

```
$ git rebase pelican_update
First, rewinding head to replay your work on top of it...
Applying: merge updates of i18n_subsites from https://github.com/getpelican/pelican-plugins/tree/master/:76a8be9 (master)
Using index info to reconstruct a base tree...
M	plugins/i18n_subsites/i18n_subsites.py
M	plugins/i18n_subsites/test_data/pelicanconf.py
Falling back to patching base and 3-way merge...
No changes -- Patch already applied.
```
closing this.